src/main/java/pl/hipotrofia/controllers/ArticleController.java

```diff
@@ -41,6 +41,7 @@ public ArticleController(ArticlesService articlesService,
         this.articleRatingsService = articleRatingsService;
     }
 
+    // thnak you :)
 
     @GetMapping("/anonymous/allToPage/{page}/{limit}/{offset}")
```

Comment:
Dla paginacji używa się:
@RequestParam zamiast @PathVariable np. /anonymous/allToPage/?page=2&limit=10&offset=0

Zamiast swoich mapowań możesz używa np. MappStruct-a - będzie mniej kodu do utrzymania. Dodanie nowego pola w JPA od razu się zmapuje do DTO.  

Zauważyłem, że rzucasz NPE w wyjątku - to jest exception runtime-owy i nie powinno się go łapac ani rzucac
np. public Measurement convertFromDto(MeasurementDto measurementDto) throws NullPointerException {
kod musi byc tak napisany aby nie rzucał takiego błędu. 

" catch (NullPointerException ex) {
            ex.printStackTrace(); -- nie loguj strackTace w logach, w ten sposób logi nie będą czytelne przy weryfikacji. Używaj np.      log.error(ex.getMessage()); log.info("") lub  log.log(Level.FATAL, ex.getMessage()); w zależności od tego co chcesz wylogowac.
            response.setStatus(404);
            response.setHeader("ERROR", "Brak danych dziecka w bazie danych");
        }
"
tak samo łapanie wyjątku 
catch (NullPointerException ex) {
         
        } catch (Exception ex) {
          
        }

Powinnny zwracac 500. 

Zamiast:
 Measurement measurement = measurementService.getById(id).orElseThrow(NegativeArraySizeException::new);

W przypadku gdy nie ma elementu do usunięcia zwracamy kod http 400.